src/test/ui/consts/control-flow/interior-mutability.rs

```diff
@@ -0,0 +1,18 @@
+#![feature(const_if_match)]
+
+use std::cell::Cell;
+
+// Ensure that *any* assignment to the return place of a value with interior mutability
```

Comment:
Do we have a test for

```
const X: Option<Cell<i32>> = {
    let mut x = Some(Cell::new(4));
    if true {
        x = None;
    }
    x
};
```

?

Reply:
Added [here](https://github.com/rust-lang/rust/pull/66507/files#diff-e2a59c2bcf3c9ce2dd92a137f9999680R16-R22).